### ISSUE TITLE
Renaming a nested type should offer to update filename using Outer.Inner.cs convention

### ISSUE THREAD

**Issue by drewnoakes** [dotnet/roslyn#60054]
**Version Used**: 17.2.0 Preview 3.0 [32308.589.main]

**Steps to Reproduce**:

Create a file `Outer.Inner.cs` containing:

```c#
partial class Outer
{
    class Inner {}
}
```

Invoke rename on the `Inner` symbol.

Notice that the rename dialog does not offer to rename the file.

![image](https://user-images.githubusercontent.com/350947/157428604-618d698a-39ca-413e-89dc-1c8e5635b035.png)

**Expected Behavior**:

Renaming `Inner` to `Foo` also renames the source file to `Outer.Foo.cs`.

Equally, renaming `Outer` to `Bar` further renames the file to `Bar.Foo.cs` (and renames any other files with the same prefix that have corresponding nested types).

Equally files nested three or more levels deep should support this `A.B.C.cs` convention.

**Actual Behavior**:

Rename for the file is disabled.

**Comment by jinujoseph** [comment]
cc @ryzngard

**Comment by sharwell** [comment]
I would not expect the name `Outer.Inner.cs` to be used. The metadata name for the nested type is `Outer+Inner`, so the file should be named `Outer+Inner.cs`.

**Comment by CyrusNajmabadi** [comment]
We should detect and support updating both cases.  Both are things that people do.  The metadata name may be Outer+Inner, but teh C#/VB names are certainly Outer.Inner, so its' totally reasonable for people to do that (and we see it in our own code and in devdiv for sure :)).
